### PR TITLE
Consistency check for incremental backup

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Allow to perform consistency check of incremental backup for backup commands that use OnlineBackup API: check will be performed if user requested verification of created backup, by specifying `verification = true` in appropriate incremental backup call.

### CODE REVIEW

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackupCommand.java [hunk 1]
```diff
@@ -203,7 +203,7 @@ public void execute( String[] args ) throws IncorrectUsage, CommandFailed
             try
             {
                 backupService.doIncrementalBackup( address.getHost(), address.getPort(),
-                        destination, timeout, config );
+                        destination, ConsistencyCheck.NONE, timeout, config );
```
Comment (tinwelint) [inline enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackupCommand.java:206]: Shouldn't we use the `check-consistency` argument for this one? Or rather introduce another one for incremental backup if we don't want to change behaviour? This is the main surface of doing a backup so it should be controllable from this surface IMHO.

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java [hunk 2]
```diff
@@ -302,7 +302,7 @@ public OnlineBackup full( String targetDirectory, boolean verification, Config t
     public OnlineBackup incremental( String targetDirectory )
     {
         outcome = new BackupService().doIncrementalBackup( hostNameOrIp, port, new File( targetDirectory ),
-                timeoutMillis, defaultConfig() );
+                getConsistencyCheck( true ), timeoutMillis, defaultConfig() );
```
Comment (tinwelint) [inline enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java]: This changes default behaviour, right? Should it?

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java [hunk 3]
```diff
@@ -324,7 +324,7 @@ public OnlineBackup incremental( String targetDirectory )
     public OnlineBackup incremental( String targetDirectory, boolean verification )
     {
         outcome = new BackupService().doIncrementalBackup( hostNameOrIp, port, new File( targetDirectory ),
-                timeoutMillis, defaultConfig() );
+                getConsistencyCheck( verification ), timeoutMillis, defaultConfig() );
```
Comment (tinwelint) [inline enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java:327]: So `verfication` wasn't quite used before? I'd guess it's fine to actually make use of it here, yeah.

**Review by tinwelint — COMMENTED** [review]
Looks good and straight forward in general. Some questions above default behaviour changing as part of this PR.

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackupCommand.java (hunk 1 above)
Comment (MishaDemianenko) [reply enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackupCommand.java:206]: On line 238 you already have all of that right?

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java (hunk 2 above)
Comment (MishaDemianenko) [reply enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java]: Right i think it's fine to do that since it's not a patch fix release and by default it's kind of reasonable to have consistent backup, also if you really want to avoid check - if that taking to long you have a way to disable it, if you still using this deprecated api.

**Comment by MishaDemianenko** [comment]
ping @tinwelint comments answered

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackupCommand.java (hunk 1 above)
Comment (spacecowboy) [reply enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackupCommand.java:206]: Yup, doing `ConsistencyCheck.NONE` here is absotely the correct thing to do.

File: enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java (hunk 2 above)
Comment (spacecowboy) [reply enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java]: Field will be upset if you change the default behavior of the old `neo4j-backup`.

Comment (MishaDemianenko) [reply enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java]: @spacecowboy you think it's better to keep then without consistency checker?

Comment (MishaDemianenko) [reply enterprise/backup/src/main/java/org/neo4j/backup/OnlineBackup.java]: agree with @spacecowboy arguments, switch to not do consistency check by default

**Comment by tinwelint** [comment]
@MishaDemianenko can you summarize here under which circumstances incremental backup will do consistency check? (because description is now out of date)

**Comment by MishaDemianenko** [comment]
@tinwelint updated, hope it's more clear now 🌈

**Comment by tinwelint** [comment]
And with that description you basically say that there's no way to trigger consistency check for incremental backup through our main surface (neo4j-admin)?

**Comment by MishaDemianenko** [comment]
@tinwelint this PR does not change anything in relation to behaviour to neo4j-admin since it already have all the options to trigger consistency check for any kind of backups if requested, you can check that in `OnlineBackupCommand`, as it says in description now tis' changes only `OnlineBackup` API

**Comment by tinwelint** [comment]
OK nice, gotcha. Thanks
